workflow-templates/im-deploy-on-prem-website.yml

```diff
@@ -0,0 +1,233 @@
+on:
+  workflow_dispatch:
+    inputs:
+      branch-tag-sha:
+        description: The branch, tag or sha of the terraform that should be deployed to the Dev environment.
+        required: true
+      environment:
+        description: The environment to Deploy To.
+        required: true
+
+env:
+  PAGERDUTY_SERVICE_IDS: '[ "${{secrets.PAGERDUTY_SERVICE_ID}}" ]' # TODO:  Add your PD Service IDs here, the svc id usually has a format of P0XXXXX
+  PAGERDUTY_WINDOW_IN_MIN: 30 # TODO:  Verify the length of your PD Maintenance Window
+  PAGERDUTY_WINDOW_DESC: 'Deploying Code to ${{ github.event.inputs.environment }} from GitHub Actions' # TODO:  Verify this PD Maintenance Window Description
+  DEPLOY_BOARD_NUM: '' # TODO: Add the automated deployment board number or remove if not using an automated deployment project board.
+  DOTNET_VERSION: '' # TODO:  Set the .net core version of your project
+  PROJECT_ROOT: '' # TODO:  set this to the folder of your web site project. For most projects this will be ./src/ProjectName
+  ARTIFACT_NAME: 'Published Site' # TODO:  Set this to the name of the artifact that shows in the workflow's summary section
+  ARTIFACT_ZIP: 'artifacts.zip'
+  DEPLOY_ZIP: 'published_site.zip'
+  IIS_SERVER_CERT_PATH: '' # TODO:  The path to the cert for the IIS server remote management
+  WEBSITE_NAME: '' # TODO:  The unique name of the site particularly for IIS server display
+  WEBSITE_HOST_HEADER: '' # TODO:  The IIS server host header that the site responds to
+  WEBSITE_CERT_PATH: '' # TODO:  The path to the private key cert for the IIS server site binding
+  WEBSITE_CERT_FRIENDLY_NAME: '' # TODO:  The friendly name encoded in the IIS private key cert
+  DEPLOYMENT_FOLDER: '' # TODO:  The local target directory for the web site files, i.e., 'c:\inetpub\website'
+
+jobs:
+  validate-input:
+    runs-on: [ubuntu-20.04]
+    steps:
+      # TODO:  Update if you don't use all these environments in the Development Subscription
+      - name: Clean Environment Input
+        id: clean-env
+        uses: im-open/map-input-action@v1.0.1
+        with:
+          input: ${{ github.event.inputs.environment }}
+          input_map: '{ "Dev": ["dev", "d", "development"], "QA": ["qa", "q"], "Stage": ["stg", "s", "stage"]  }'
+          error_on_no_match: true
+          custom_error_message: 'The environment must be Dev, QA or Stage'
+
+  tech-approval:
+    if: ${{ github.event.inputs.environment }} != "DEV" && ${{ github.event.inputs.environment }} != "QA"
+    runs-on: [ubuntu-20.04]
+    environment: tech-approve
+
+  qa-approval:
+    if: ${{ github.event.inputs.environment }} != "DEV" && ${{ github.event.inputs.environment }} != "QA"
+    needs: tech-approval
+    runs-on: [ubuntu-20.04]
+    environment: qa-approve
+
+  stakeholder-approval:
+    if: ${{ github.event.inputs.environment }} == "PROD"
+    needs: [tech-approval, qa-approval]
+    runs-on: [ubuntu-20.04]
+    environment: stakeholder-approve
```

Comment:
The other workflows that do this kind of thing split it in to two workflows, one for deploying to dev/qa/stage and one for deploying to prod/stage (stage is different depending on team).  If you aren't going to split this up I would definitely add some TODOs above each of those approval environments recommending it for certain scenarios and removing it for others.

Reply:
Just to follow up with this, I'm condensing prod/dev workflows into one.